### PR TITLE
Adding support for `all time` in filter for paginated searches

### PR DESCRIPTION
**Pull request by janheise**
## Description

When selecting `All time` in the search filter -> "Created At" section for paginated searches (e.g. Streams overview or Dashboards), we get the following Query Param: `created_at:><` which fails to parse. This PR fixes the issue.

fixes https://github.com/Graylog2/graylog2-server/issues/23634

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have requested a documentation update.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/database/filtering/inmemory/SingleFilterParser.java [hunk 1]
```diff
@@ -51,7 +51,10 @@ public Filter parseSingleExpression(final String filterExpression, final List<En
 
         final SearchQueryField.Type fieldType = attributeMetaData.type();
         if (isRangeValueExpression(valuePart, fieldType)) {
-            if (valuePart.startsWith(RANGE_VALUES_SEPARATOR)) {
+            if (valuePart.equals(RANGE_VALUES_SEPARATOR)) {
+                // could probably also return an empty BSON here, but just for consistency:
+                return new RangeFilter(attributeMetaData.id(), null, null);
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog2/database/filtering/inmemory/SingleFilterParser.java:56]: It does not have to be done in this PR, but I guess that RangeFilter class would need a unit test as well, checking that `toBson()` method returns proper BSONs, especially for filters with null values.

**Review by luk-kaminski — APPROVED** [review]
LGTM, thanks!
